packages/components/pages/f-mfa/src/components/_tests/Mfa.test.js

```diff
@@ -267,5 +267,28 @@ describe('Mfa', () => {
                 expect(pushEventSpy).toMatchSnapshot();
             });
         });
+
+        it('should not make another API call if `isSubmitting` is already true', async () => {
+            // Arrange
+            mockPostValidateMfaToken = jest.fn();
+            wrapper = await mountSut({
+                propsData: {
+                    ...sutProps,
+                    code: 'XYZ1234'
+                }
+            });
+            await wrapper.setData({
+                isSubmitting: true
+            });
+            pushEventSpy.mockClear();
+
+            // Act
+            await wrapper.vm.onFormSubmit();
+
+            // Assert
+            expect(mockPostValidateMfaToken).not.toHaveBeenCalled();
+            expect(pushEventSpy).not.toHaveBeenCalled();
+            expect(wrapper.vm.isSubmitting).toBe(true);
```

Comment:
I don't think you need this line.

Reply:
will fix on next pr, nw

Reply:
@kevinrodrigues @oliversweb This line is helpful, it checks that we haven't re-enabled the button. It's relevant because of the `finally` block, which should be skipped over if the button is clicked for a second time, and should only be called when the first request comes back.

Reply:
Ok i see that makes sense now, I hadn't realised it was related to the early return. 👍  I would suggest splitting tests out with a description, that way it's a little easier to understand the unit of test.

```javascript
describe('And "isSubmitting" is truthy', () => ..
   it('should exit early to prevent multiple requests')
````